### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by waynarbocangel**
Check this PR as much as you feel comfortable with. It changes an absurd amount of files because it integrates mostly everything into our UI. Most of it should be either CRUD in the src/front-end/localStorage folder with respective tests (all pass but we need more tests that Richard To will finish soon, we are pushing as advised by Sanat). This also merges changes in main to the front-end meaning after merging this should be ready to push to prod. The rest of the changes are integration, they handle data display and some css edits here and there. Pretty major so take your time.

### CODE REVIEW

**Comment by forschoolandnothingelse** [comment]
Waynar, none of your commit messages follow guidelines.

**Comment by waynarbocangel** [comment]
> Waynar, none of your commit messages follow guidelines.

A lot of them do, but I do apologize for the ones that don't. All the last ones I pushed were angry attempts to make GH Actions run the tests it should be running. For those ones I just got frustrated and didn't really follow the guidelines so I'm sorry. Essentially everything has been edited to some degree too so the folder convention was hard to follow specially after squashing some commits. Sorry, lmk if I can help clarify any of them.

File: src/back-end/index.js [hunk 1]
```diff
@@ -42,6 +43,9 @@ mongoose.set("useCreateIndex", true);
 
 /* Listen for connection on port 2827 */
 app.listen("2827", () => {
+    console.log("Environment");
+    let db = process.env.DB;
+    console.log(db);
```
Comment (Ehson12) [inline src/back-end/index.js:48]: Maybe remove this console.log?

Comment (ehsonp) [inline src/back-end/index.js:48]: Maybe remove this console.log?

File: src/front-end/__tests__/coverage/coverage-final.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (ehsonp) [inline src/front-end/__tests__/coverage/coverage-final.json:1]: maybe remove file?

**Review by ehsonp — APPROVED** [review]
Alright i would normally disapprove of such a large commit, it makes it very very difficult to actually review it well, its daunting. 

I looked through most of the code and it looked okay, 
I tested the code and it ran half the things well.

the list of broken things
1. text editor cant save the blocks
2. index page looks horrible (i know darins working on redesign) 
3. missing functionality with some buttons  (namely edit)
4. calendar is a tad wonky.

I will however still approve this so we can actually make these changes and fixes in smaller PR's.

Approved

**Review by e5chang — APPROVED** [review]
I agree, this is a large commit of 285 files changed. Next time we should probably get smaller features to work and merge them in smaller PRs to review more detailed. There are a lot of small bugs and index does not look like I expected it to. Otherwise, it should be okay to merge and fix these issues after

**Comment by rltoSD** [comment]
Just finished reviewing this; separating this up into separate PRs might've been smarter but looks fine to me
